pageserver/src/layered_repository.rs

```diff
@@ -376,19 +387,21 @@ impl LayeredRepository {
 
         // fsync the parent directory to ensure the directory entry is durable
         if first_save {
-            let timeline_dir = File::open(&timeline_path)?;
-            timeline_dir.sync_all()?;
+            if let Some(timeline_path) = path.parent() {
```

Comment:
Why this change?

Reply:
Mainly purism, can revert it back.

Before, `timeline_path` was constructed only to create the path to metadata file.
Later, `timeline_path` was used to sync the FS metadata after our metadata file was written.
So `timeline_path` was explicitly used as a parent, assuming that it never changes.

Now I have a method to get the metadata path directly and I can move away from the "metadata file is stored inside `timeline_path` directory" hardcoded knowledge.

I can do `.parent().unwrap()` as a middle ground to emphasize that we're dealing with correctly created metadata, but not fond of potential panics myself.